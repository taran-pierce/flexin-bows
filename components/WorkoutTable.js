import WorkoutTableHeader from './WorkoutTableHeader'
import Link from 'next/link'

const WorkoutTable = (props) => {
  const days = props.plans.days
  
  return (
    <div className={`workout-table`}>
      <WorkoutTableHeader days={days} />
      <style jsx>{`
        table {
          border-collapse: collapse;
          text-align: left;
          width: 100%;
        }
        
        td, th {
          padding: .5rem 1rem;
          border-right: 1px solid black;
          border-bottom: 1px solid black;
        }
        
        th {
          border-bottom-width: 2px;
        }
        
        th:last-child,
        td:last-child {
          border-right: none;
        }
        
        .workout-table {
          border: 1px solid black;
          border-bottom: none;
        }
      `}</style>
      
      {days.map( (day) => (
        <table key={day.title}>
          <thead>
          <tr>
            <th>Body Part</th>
            <th>Exercise</th>
            <th>Sets</th>
            <th>Reps</th>
          </tr>
        </thead>
          <tbody>
            {day.bodyParts.map( (part) => (
              <tr key={part.name}>
                <td>{part.name}</td>
                <td>
                  {part.exercises.map( (workout) => (
                    <p key={workout.name}>
                      <Link href={`#`}>
                        <a>{workout.name}</a>
                      </Link>
                    </p>
                  ))}
                </td>
                <td>{part.sets}</td>
                <td>{part.reps}</td>
              </tr>
            ))}
          </tbody>
        </table>
      ))}
    </div>
  )
}

export default WorkoutTable