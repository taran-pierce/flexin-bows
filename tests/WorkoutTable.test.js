import React from 'react';
import WorkoutTable from '../components/WorkoutTable';
import renderer from 'react-test-renderer';
import data from '../data/programs/better-body.json'

test('Does WorkoutTable render correctly?', () => {
  const plans = data.workout
  
  const component = renderer.create(
    <WorkoutTable plans={plans} />,
  );
  
  // verify component compiles correctly
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
