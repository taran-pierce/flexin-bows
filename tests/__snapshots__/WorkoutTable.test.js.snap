// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does WorkoutTable render correctly? 1`] = `
<div
  className="jsx-1390947089 workout-table"
>
  <div
    className="jsx-92330898"
  >
    <ul
      className="jsx-92330898"
    >
      <li
        className="jsx-92330898"
      >
        <a
          className="jsx-92330898"
          data-id="better-body-M-W-F"
          href="#"
        >
          M-W-F
        </a>
      </li>
    </ul>
  </div>
  <div
    className="jsx-1390947089 "
  >
    <table
      className="jsx-1390947089 hidden"
    >
      <thead
        className="jsx-1390947089"
      >
        <tr
          className="jsx-1390947089"
        >
          <th
            className="jsx-1390947089"
          >
            Body Part
          </th>
          <th
            className="jsx-1390947089"
          >
            Exercise
          </th>
          <th
            className="jsx-1390947089"
          >
            Sets
          </th>
          <th
            className="jsx-1390947089"
          >
            Reps
          </th>
        </tr>
      </thead>
      <tbody
        className="jsx-1390947089"
      >
        <tr
          className="jsx-1390947089"
        >
          <td
            className="jsx-1390947089"
          >
            Chest
          </td>
          <td
            className="jsx-1390947089"
          >
            <p
              className="jsx-1390947089"
            >
              <a
                className="jsx-1390947089"
                data-file="bench-press-horizontal"
                href="#"
              >
                Bench Press
              </a>
            </p>
          </td>
          <td
            className="jsx-1390947089"
          >
            1 - 2
          </td>
          <td
            className="jsx-1390947089"
          >
            10 - 15
          </td>
        </tr>
        <tr
          className="jsx-1390947089"
        >
          <td
            className="jsx-1390947089"
          >
            Back
          </td>
          <td
            className="jsx-1390947089"
          >
            <p
              className="jsx-1390947089"
            >
              <a
                className="jsx-1390947089"
                data-file="shoulder-extension"
                href="#"
              >
                Seated Lat Rows
              </a>
            </p>
          </td>
          <td
            className="jsx-1390947089"
          >
            1 - 2
          </td>
          <td
            className="jsx-1390947089"
          >
            10 - 15
          </td>
        </tr>
        <tr
          className="jsx-1390947089"
        >
          <td
            className="jsx-1390947089"
          >
            Shoulders
          </td>
          <td
            className="jsx-1390947089"
          >
            <p
              className="jsx-1390947089"
            >
              <a
                className="jsx-1390947089"
                data-file="cross-over-rear-deltiod"
                href="#"
              >
                Crossover Seated Rear Deltoid Rows
              </a>
            </p>
          </td>
          <td
            className="jsx-1390947089"
          >
            1 - 2
          </td>
          <td
            className="jsx-1390947089"
          >
            10 - 15
          </td>
        </tr>
        <tr
          className="jsx-1390947089"
        >
          <td
            className="jsx-1390947089"
          >
            Arms
          </td>
          <td
            className="jsx-1390947089"
          >
            <p
              className="jsx-1390947089"
            >
              <a
                className="jsx-1390947089"
                data-file="bicep-curl"
                href="#"
              >
                Standing Biceps Curl
              </a>
            </p>
            <p
              className="jsx-1390947089"
            >
              <a
                className="jsx-1390947089"
                data-file="triceps-pushdown"
                href="#"
              >
                Triceps Pushdown
              </a>
            </p>
          </td>
          <td
            className="jsx-1390947089"
          >
            1 - 2
          </td>
          <td
            className="jsx-1390947089"
          >
            10 - 15
          </td>
        </tr>
        <tr
          className="jsx-1390947089"
        >
          <td
            className="jsx-1390947089"
          >
            Legs
          </td>
          <td
            className="jsx-1390947089"
          >
            <p
              className="jsx-1390947089"
            >
              <a
                className="jsx-1390947089"
                data-file="calf-raise"
                href="#"
              >
                Calf Raise
              </a>
            </p>
            <p
              className="jsx-1390947089"
            >
              <a
                className="jsx-1390947089"
                data-file="leg-press"
                href="#"
              >
                Leg Press
              </a>
            </p>
          </td>
          <td
            className="jsx-1390947089"
          >
            1 - 2
          </td>
          <td
            className="jsx-1390947089"
          >
            10 - 15
          </td>
        </tr>
        <tr
          className="jsx-1390947089"
        >
          <td
            className="jsx-1390947089"
          >
            Trunk
          </td>
          <td
            className="jsx-1390947089"
          >
            <p
              className="jsx-1390947089"
            >
              <a
                className="jsx-1390947089"
                data-file="trunk-rotation"
                href="#"
              >
                Trunk Rotation
              </a>
            </p>
            <p
              className="jsx-1390947089"
            >
              <a
                className="jsx-1390947089"
                data-file="abdominal-crunch"
                href="#"
              >
                Seated Abdominal Crunch
              </a>
            </p>
          </td>
          <td
            className="jsx-1390947089"
          >
            1 - 2
          </td>
          <td
            className="jsx-1390947089"
          >
            10 - 15
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
